Library/Homebrew/cmd/install.rb

```diff
@@ -24,8 +24,13 @@
 #:    `gcc-4.2` for Apple's GCC 4.2, or `gcc-4.9` for a Homebrew-provided GCC
 #:    4.9.
 #:
-#:    If `--build-from-source` is passed, compile from source even if a bottle
-#:    is provided for <formula>.
+#:    If `--build-from-source` or `-s` is passed, compile the specified <formula> from
+#:    source even if a bottle is provided. Dependencies will still be installed
+#:    from bottles if they are available.
+#:
+#:    If `--build-all-from-source` or `-S` is passed, or HOMEBREW_BUILD_FROM_SOURCE is set,
```

Comment:
Doesn't affect the code difficulty. I was including it because I think it's more convenient than the environment variable: it could be included in our tab completion definitions, and if a user is doing a `-s` install and realizes they need to switch to `-S`, or vice versa, it's easy to edit the command line by just adding or removing the `all-` or switching case, instead of having to remember and add the whole environment variable.